### PR TITLE
fix: Several minor issues in corellium scope

### PR DESCRIPTION
**Pull request by jan-gogo**
Related to #1802

This is a bunch of several minor changes cherry-picked from #1897 for splitting code review into smaller parts.

* Normalize output lines from Corellium console + unit test
* Convert API interfaces to functional
    * Update adapters implementation to use SAM conversion
* Rename API scope TestApk to Apk and expand the interface to meet domain requirements
* Fix log parser issues
* Add doc for sharding structures
* Add `am instrument` status codes constants

Merging this one will make the #1897 thinner.

### CODE REVIEW

File: corellium/adapter/src/test/kotlin/flank/corellium/adapter/Example.kt [hunk 1]
```diff
@@ -63,14 +64,14 @@ fun main() {
     runBlocking {
 
         println("* Authorizing")
-        api.authorize(credentials)
+        api.authorize(credentials).join()
 
         println("* Invoking devices")
-        val ids = api.invokeAndroidDevices(AndroidInstance.Config(2)).toMutableList()
+        val ids = api.invokeAndroidDevices(AndroidInstance.Config(2)).toList().toMutableList()
```
Comment (Sloox) [inline corellium/adapter/src/test/kotlin/flank/corellium/adapter/Example.kt:70]: .toList().toMutableList()
Is there a reason for this?
If this is already a collection its not needed.

File: corellium/log/src/main/kotlin/flank/corellium/log/Internal.kt [hunk 2]
```diff
@@ -67,12 +67,16 @@ internal fun Flow<List<Line>>.parseChunks(): Flow<Chunk> = map { group ->
...
-                add(text)
-                map[key] = reversed()
+                add(text.trim())
+                map[key.trim()] = reversed()
                 clear()
             }
     }
-    Chunk(code.prefix!!, code.text.toInt(), map)
+    Chunk(
+        type = code.prefix!!,
```
Comment (Sloox) [inline corellium/log/src/main/kotlin/flank/corellium/log/Internal.kt:76]: double bang usually is a bad thing :(

**Review by Sloox — APPROVED** [review]
Minor comments :)

File: corellium/log/src/main/kotlin/flank/corellium/log/Internal.kt (hunk 2 above)
Comment (jan-gogo) [reply corellium/log/src/main/kotlin/flank/corellium/log/Internal.kt]: Fixed

**Review by adamfilipow92 — APPROVED** [review]

**Pull request merged**